crates/runtime/src/component/bus1_0_0.rs

```diff
@@ -0,0 +1,60 @@
+//! Compatibility implementation of the `wasmcloud:bus/lattice@1.0.0` interface
+use super::{Ctx, Handler, TableResult};
+
+use crate::capability::bus1_0_0::lattice;
+
+use std::sync::Arc;
+
+use anyhow::Context as _;
+use async_trait::async_trait;
+use wasmcloud_core::CallTargetInterface;
+use wasmtime::component::Resource;
+
+#[async_trait]
+/// `wasmcloud:bus/lattice@1.0.0` implementation
+pub trait Bus {
+    /// Set the link name to use for a given list of interfaces
+    async fn set_link_name(&self, link_name: String, interfaces: Vec<Arc<CallTargetInterface>>);
+}
+
+#[async_trait]
+impl<H: Handler> lattice::Host for Ctx<H> {
+    async fn set_link_name(
+        &mut self,
+        link_name: String,
+        interfaces: Vec<Resource<Arc<CallTargetInterface>>>,
+    ) -> anyhow::Result<()> {
+        let interfaces = interfaces
+            .into_iter()
+            .map(|interface| self.table.get(&interface).cloned())
+            .collect::<TableResult<_>>()
+            .context("failed to convert call target interfaces")?;
+        self.handler
+            .set_link_name(link_name, interfaces)
+            .await
+            .map(|_| ())
```

Comment:
nit; I usually find that splitting such occurrences into `?` followed by `Ok(())` is:
1. clearer/easier to understand
2. ends up with less LoC due to `rustfmt` formatting
3. more error-prone in case the inner `Result::Ok` is not what one expected

```suggestion
        self.handler
            .set_link_name(link_name, interfaces)
            .await?;
        Ok(())
```

I'm fine either way though

Reply:
I could go either way here, changed to this just to simplify